### PR TITLE
V5 - allow middleware use

### PR DESCRIPTION
**Pull request by juliangut**
Pass Response object to `respondToRequest` to be composed with result of auth check instead of creating a new Response
`__invoke` method to allow to be used as middleware

### CODE REVIEW

File: src/Exception/OAuthServerException.php [hunk 1]
```diff
@@ -221,12 +225,13 @@ public function generateHttpResponse()
...
         }
 
-        $response = new Response(
-            'php://memory',
-            $this->getHttpStatusCode(),
-            $headers
-        );
-        $response->getBody()->write(json_encode($payload));
+        foreach ($headers as $header => $content) {
+            $response->withHeader($header, $content);
```
Comment (andersonamuller) [inline src/Exception/OAuthServerException.php]: You forgot the immutability: 

``` php
$response = $response->withHeader($header, $content);
```

File: src/Exception/OAuthServerException.php [hunk 2]
```diff
@@ -221,12 +225,13 @@ public function generateHttpResponse()
...
+        }
+
+        $response
```
Comment (andersonamuller) [inline src/Exception/OAuthServerException.php]: You forgot the immutability: 

``` php
$response = $response
              ->withStatus($this->getHttpStatusCode())
              ->getBody()->write(json_encode($payload));
```

or 

```
return $response
          ->withStatus($this->getHttpStatusCode())
          ->getBody()->write(json_encode($payload));
```

Comment (alexbilbie) [inline src/Exception/OAuthServerException.php]: Good spot!

**Comment by alexbilbie** [comment]
Thanks @juliangut, will fix the immutability issues highlighted by @andersonamuller

**Pull request merged**